client-v2/src/components/FrontsEdit/FrontContainer.tsx

```diff
@@ -196,6 +246,49 @@ class Fronts extends React.Component<FrontsComponentProps, ComponentState> {
       </SingleFrontContainer>
     );
   }
+
+  private renameFront = () => {
+    this.setState({
+      frontNameValue: this.getTitle() || '',
+      editingFrontName: true
+    });
+  };
+
+  private getTitle = () => {
+    const { selectedFront } = this.props;
+
+    if (selectedFront) {
+      if (selectedFront.metadata && selectedFront.metadata.nameOverride) {
+        return selectedFront.metadata.nameOverride;
+      } else {
+        return startCase(
+          this.props.selectedFront.displayName || this.props.selectedFront.id
```

Comment:
minor: we've destructured above, so can reuse `selectedFront`

```suggestion
          selectedFront.displayName || selectedFront.id
```